tiledb/sm/group/group_details.h

```diff
@@ -0,0 +1,246 @@
+   *
+   * @param group_member_uri group member uri
+   * @param relative is this URI relative
+   * @param name optional name for member
+   * @return Status
+   */
+  Status mark_member_for_addition(
+      const URI& group_member_uri,
+      const bool& relative,
+      std::optional<std::string>& name,
+      StorageManager* storage_manager);
+
+  /**
+   * Remove a member from a group, this will be flushed to disk on close
+   *
+   * @param uri of member to remove
+   * @return Status
+   */
+  Status mark_member_for_removal(const URI& uri);
+
+  /**
+   * Remove a member from a group, this will be flushed to disk on close
+   *
+   * @param uri of member to remove
+   * @return Status
+   */
+  Status mark_member_for_removal(const std::string& uri);
+
+  /**
+   * Get the vector of members to modify, used in serialization only
+   * @return members_to_modify
+   */
+  const std::vector<tdb_shared_ptr<GroupMember>>& members_to_modify() const;
+
+  /**
+   * Get the unordered map of members
+   * @return members
+   */
+  const std::unordered_map<std::string, tdb_shared_ptr<GroupMember>>& members()
+      const;
+
+  /**
+   * Add a member to a group
+   *
+   * @param group_member to add
+   * @return void
+   */
+  void add_member(const tdb_shared_ptr<GroupMember>& group_member);
+
+  /**
+   * Delete a member from the group
+   *
+   * @param group_member
+   */
+  void delete_member(const tdb_shared_ptr<GroupMember>& group_member);
+
+  /**
+   * Serializes the object members into a binary buffer.
+   *
+   * @param buff The buffer to serialize the data into.
+   * @param version The format spec version.
+   * @return Status
+   */
+  virtual void serialize(Serializer& serializer);
+
+  /**
+   * Returns a Group object from the data in the input binary buffer.
+   *
+   * @param deserializer The buffer to deserialize from.
+   * @param version The format spec version.
+   * @return Status and Attribute
+   */
+  static std::optional<tdb_shared_ptr<GroupDetails>> deserialize(
+      Deserializer& deserializer, const URI& group_uri);
+
+  /**
+   * Returns a Group object from the data in the input binary buffer.
+   *
+   * @param deserializers List of buffers for each details file to deserialize
+   * from.
+   * @param version The format spec version.
+   * @return Status and Attribute
+   */
+  static std::optional<tdb_shared_ptr<GroupDetails>> deserialize(
+      const std::vector<shared_ptr<Deserializer>>& deserializer,
+      const URI& group_uri);
+
+  /** Returns the group URI. */
+  const URI& group_uri() const;
+
+  /**
+   * Have changes been applied to a group in write mode
+   * @return changes_applied_
+   */
+  bool changes_applied() const;
+
+  /**
+   * Set changes applied, only used in serialization
+   * @param changes_applied should changes be considered to be applied? If so
+   * then this will enable writes from a deserialized group
+   *
+   */
+  void set_changes_applied(bool changes_applied);
```

Comment:
Should not be public, IMO make protected if we anticipate needing it (ideally let's avoid adding setters for internal state whenever we can).